server/channels/app/draft.go

```diff
@@ -61,6 +61,10 @@ func (a *App) UpsertDraft(c *request.Context, draft *model.Draft, connectionID s
 		return nil, model.NewAppError("CreateDraft", "app.draft.save.app_error", nil, nErr.Error(), http.StatusInternalServerError)
 	}
 
+	if draft.Message == "" {
+		a.Srv().Store().Draft().Delete(draft.UserId, draft.ChannelId, draft.RootId)
+	}
+
```

Comment:
Should we check this before the `Upsert` above?

Reply:
I'm not sure if it matters, because if the above query fails (maybe due to locking) we wouldn't want to try the delete. I assume we just want to make sure that we don't save empty messages in the DB without also having marked them deleted.

Reply:
I'd say in case of deleting, only delete, not upsert and delete (one vs 2 DB calls). Doesn't feel right upserting only to delete it later.

Reply:
@harshilsharma63 Wouldn't that mean that there would still be a message in the draft on the DB side? Not sure if that would matter or not.

Reply:
@harshilsharma63 Wouldn't that mean that there would still be a message in the draft on the DB side? Not sure if that would matter or not.